guidance/models/_azureai_studio.py

```diff
@@ -0,0 +1,161 @@
+import hashlib
+import json
+import pathlib
+import urllib.request
+
+import diskcache as dc
+import platformdirs
+
+from ._model import Chat
+from ._grammarless import GrammarlessEngine, Grammarless
+
+
+class AzureAIStudioChatEngine(GrammarlessEngine):
+    def __init__(
+        self,
+        *,
+        tokenizer,
```

Comment:
I never try setting the tokeniser, and it appears that it eventually defaults to GPT2. I don't quite see why a remote model like this would even need it.

Reply:
OK, so theoretically for token healing. However, I have a feeling that trying to figure out what tokeniser to use will be an exercise in fragility.